src/libraries/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -9,227 +9,316 @@ internal struct FileStatus
     {
         private const int NanosecondsPerTick = 100;
 
-        // The last cached stat information about the file
-        private Interop.Sys.FileStatus _fileStatus;
+        // Exists as of the last refresh
+        private bool _exists;
 
-        // -1 if _fileStatus isn't initialized, 0 if _fileStatus was initialized with no
-        // errors, or the errno error code.
-        private int _fileStatusInitialized;
+        // The last cached lstat information about the file
+        private Interop.Sys.FileStatus _mainCache;
 
-        // We track intent of creation to know whether or not we want to (1) create a
-        // DirectoryInfo around this status struct or (2) actually are part of a DirectoryInfo.
-        internal bool InitiallyDirectory { get; private set; }
+        // The last cached stat information about the file
+        // Refresh only collects this if lstat determines the path is a symbolic link
+        private Interop.Sys.FileStatus _secondaryCache;
 
```

Comment:
after reading this comment I think that the `_mainCache` should be called `_fileCache` and `_secondaryCache` should be called `_symlinkCache`

Reply:
Sure, I can rename them.